### PR TITLE
Esc 647

### PR DESCRIPTION
**Pull request by sam-miele**

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/actions/EscInitialRequestActionBuilder.scala [hunk 1]
```diff
@@ -38,7 +40,7 @@ class EscInitialRequestActionBuilder @Inject() (
   val config: Configuration,
   val env: Environment,
   val authConnector: AuthConnector,
-  mcc: ControllerComponents
+  mcc: ControllerComponents,
```
Comment (sam-miele) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/actions/EscInitialRequestActionBuilder.scala:43]: I should remove this

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala [hunk 2]
```diff
@@ -146,14 +153,107 @@ class UndertakingController @Inject() (
...
+    }
+  }
+
+  def getVerifyEmail(pendingVerificationId: String): Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    implicit val eori: EORI = request.eoriNumber
+    store.get[UndertakingJourney].flatMap {
+      case Some(journey) =>
+        for {
+          e <- emailVerificationService.verifyVerificationRequest(request.eoriNumber, pendingVerificationId)
+          wasSuccessful = e.getMatchedCount > 0
```
Comment (carwynellis) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala:236]: Are we going to have mulitple occurrences of the same email address? I figured we could use `eori, email` as a unique key but I could be wrong.

File: test/uk/gov/hmrc/eusubsidycompliancefrontend/services/EmailVerificationServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      }
+    }
+
+    "verifyVerificationRequest is called" must {
+
+      "verify record" when {
+        "success" in {
+          Await.result(repository.put(eori1, unverifiedVerificationRequest.copy(pendingVerificationId = "pending".some)), Duration(10, "seconds"))
+          service.verifyVerificationRequest(eori1, "pending")
+          service.getEmailVerification(eori1).futureValue shouldBe unverifiedVerificationRequest.copy(verified = true.some, pendingVerificationId = "pending".some).some
```
Comment (carwynellis) [inline test/uk/gov/hmrc/eusubsidycompliancefrontend/services/EmailVerificationServiceSpec.scala:120]: If I run the entire suite locally I get a failure on this test as follows

```
[info]   when verifyVerificationRequest is called
[info]     must verify record
[info]     - when success *** FAILED ***
[info]       None was not equal to Some(EmailVerificationState(unverified@something.com,Some(pending),Some(true))) (EmailVerificationServiceSpec.scala:120)
```

It seems to work if I run the test in isolation. Might be worth looking into if you get this behaviour. Likely I'm missing something somewhere!

**Review by carwynellis — CHANGES_REQUESTED** [review]
There's a few minor things here you might want to look at.

Also doing a `clean compile test:compile`, there are 44 warnings that ought to be cleaned up.

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala [hunk 4]
```diff
@@ -146,14 +153,107 @@ class UndertakingController @Inject() (
...
+            case Some(verifiedEmail) => {
+              Ok(confirmEmailPage(optionalEmailForm, EmailAddress(verifiedEmail.email), routes.UndertakingController.getSector().url)).toFuture
+            }
+            case None => emailService.retrieveEmailByEORI(request.eoriNumber) map { response =>
+              response.emailType match {
+                case EmailType.VerifiedEmail => {
+                  Ok(confirmEmailPage(optionalEmailForm, response.emailAddress.get, routes.UndertakingController.getSector().url))
+                }
+                case _ => {
+                  Ok(inputEmailPage(emailForm, routes.UndertakingController.getSector().url))
```
[the remark this answers is not in the document]
Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala:173]: Yeah! So the outer optional is the "us" verified email. So these two branches under the None case (line 167) are asking "OK, we have no verified email in our store. Do you have one in CDS? If so we'll provide the default for the input".

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala (hunk 2 above)
Comment (sam-miele) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala:236]: We decided that it's just one verified email per EORI. Good catch though!

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (carwynellis) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala:173]: Gotcha.

**Review by carwynellis — APPROVED** [review]
LGTM 👍 

I think there may still be some compiler warnings to squish first though.

**Pull request merged**